mxresolv/mxresolv.go

```diff
@@ -145,3 +151,6 @@ func cacheAndReturn(hostname string, mxHosts []string, implicit bool, err error)
 	lookupResultCache.AddWithTTL(hostname, lookupResult{mxHosts, implicit, err}, cacheTTL)
 	return mxHosts, implicit, err
 }
+
+//go:linkname lookupMX net.(*Resolver).lookupMX
```

Comment:
Would recommend a bit of documentation around why this approach was used, such as your explanation in #pipeline-dev.